datafusion-postgres/src/datatypes.rs

```diff
@@ -361,6 +372,17 @@ fn encode_value(
                 DataType::UInt64 => encoder.encode_field(&get_u64_list_value(arr, idx))?,
                 DataType::Float32 => encoder.encode_field(&get_f32_list_value(arr, idx))?,
                 DataType::Float64 => encoder.encode_field(&get_f64_list_value(arr, idx))?,
+                DataType::Decimal128(_, s) => {
+                    let list_arr = arr.as_any().downcast_ref::<ListArray>().unwrap().value(idx);
+                    let value: Vec<_> = list_arr
+                        .as_any()
+                        .downcast_ref::<Decimal128Array>()
+                        .unwrap()
+                        .iter()
+                        .map(|v| Decimal::from_i128_with_scale(v.unwrap(), *s as u32))
```

Comment:
Is it 100% safe to use `unwrap` here

Reply:
you are right, changed to handle None. Also improve error handling